streaming/java/streaming-api/src/main/java/org/ray/streaming/api/stream/DataStream.java

```diff
@@ -124,19 +136,44 @@ public DataStream(DataStream input, StreamOperator streamOperator) {
    * @return This stream.
    */
   public DataStream<T> partitionBy(Partition<T> partition) {
-    this.partition = partition;
+    checkPartitionCall();
+    super.setPartition(partition);
```

Comment:
Unnecessary `super` it should be.

Reply:
As a further reviewing it seems could be `return setPartition(partition)`. The point is that we shouldn't call super.otherMethod in thisMethod, otherwise other contributors will get confused why we don't call this.otherMethod.

Reply:
Good catch, just fixed it